src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Type.hs

```diff
@@ -154,13 +155,9 @@ data ConsolidatedRowReferences b a = ConsolidatedRowReferences
   , theRowWidth :: RowWidth
   }
 
--- | This wrapper facilitates naming the original structure
--- (i.e. the "payload" for recognition)
--- for the purpose of both UI display and internal uniqueness,
--- while remaining agnostic to its internals.
-data NamedOriginal b = NamedOriginal
-  { getName :: StructureName
-  , orig :: NamedGrid b
+data ExtractedArea b a = ExtractedArea
+  { originalItem :: NamedArea b
+  , extractedGrid :: NonEmptyGrid (AtomicKeySymbol a)
```

Comment:
This new `ExtractedArea` type is what propagates the original `b` through the algorithm.